### PR TITLE
Make the example use the new stop semantic

### LINKED ISSUE #4: refactor shutdown sequence to use plugin.stop
Change shutdown sequence to be triggered by plugin.stop instead of ShutdownSignal exception.
Also remove any calls to: shutdown, finished, finished?, running? or terminating?
This depends on https://github.com/elastic/logstash/pull/3895

### PR DESCRIPTION
**Pull request by untergeek**
closes #4

### CODE REVIEW

File: lib/logstash/inputs/example.rb [hunk 1]
```diff
@@ -28,11 +28,17 @@ def register
   end # def register
 
   def run(queue)
+    @thread = Thread.current
+
     Stud.interval(@interval) do
       event = LogStash::Event.new("message" => @message, "host" => @host)
       decorate(event)
       queue << event
+      break if stop?
```
Comment (purbon) [inline lib/logstash/inputs/example.rb]: i wonder if break is necessary here? what are the benefits of having it (I mean educationally for people seeing this examples?)

When using interval I understand this break is not providing any info, what we could for example add is a comment indicating that if there are slow operations we could do something like this. What do you think?

**Comment by purbon** [comment]
@untergeek small more conceptual doubt, for the rest all good.

File: lib/logstash/inputs/example.rb (hunk 1 above)
Comment (ph) [inline lib/logstash/inputs/example.rb]: @purbon is right here, there is no need to break out of the interval, `Stud.stop!` will break it on the next loop.

File: lib/logstash/inputs/example.rb [hunk 2]
```diff
@@ -28,11 +28,17 @@ def register
   end # def register
 
   def run(queue)
+    @thread = Thread.current
```
Comment (ph) [inline lib/logstash/inputs/example.rb:36]: This is not obvious for new plugin developers. I think its worth to add why we need to keep a reference to the current thread inside the `#run` maybe a small explanation even if its related to the `Stud.interval` case.

**Comment by ph** [comment]
@untergeek Thanks for updating this and +1 for specifying compatible version for stud and devutils.

**Comment by ph** [comment]
LGTM

**Pull request closed without merging**